dpms/dpms.go

```diff
@@ -43,6 +43,102 @@ const (
 	DPMSModeOff     = 3
 )
 
+const (
+	EventMaskInfoNotify = 1
+)
+
+// InfoNotify is the event number for a InfoNotifyEvent.
+const InfoNotify = 0
+
+type InfoNotifyEvent struct {
+	Sequence uint16
+	// padding: 2 bytes
+	Timestamp  xproto.Timestamp
+	PowerLevel uint16
+	State      bool
+	// padding: 21 bytes
+}
+
+// InfoNotifyEventNew constructs a InfoNotifyEvent value that implements xgb.Event from a byte slice.
+func InfoNotifyEventNew(buf []byte) xgb.Event {
+	v := InfoNotifyEvent{}
+	b := 1 // don't read event number
+
+	b += 2 // padding
+
+	v.Sequence = xgb.Get16(buf[b:])
+	b += 2
+
+	v.Timestamp = xproto.Timestamp(xgb.Get32(buf[b:]))
+	b += 4
+
+	v.PowerLevel = xgb.Get16(buf[b:])
+	b += 2
+
+	if buf[b] == 1 {
+		v.State = true
+	} else {
+		v.State = false
+	}
+	b += 1
+
+	b += 21 // padding
+
+	return v
+}
+
+// Bytes writes a InfoNotifyEvent value to a byte slice.
+func (v InfoNotifyEvent) Bytes() []byte {
+	buf := make([]byte, 32)
+	b := 0
+
+	// write event number
+	buf[b] = 0
+	b += 1
+
+	b += 2 // padding
+
+	b += 2 // skip sequence number
+
+	xgb.Put32(buf[b:], uint32(v.Timestamp))
+	b += 4
+
+	xgb.Put16(buf[b:], v.PowerLevel)
+	b += 2
+
+	if v.State {
+		buf[b] = 1
+	} else {
+		buf[b] = 0
+	}
+	b += 1
+
+	b += 21 // padding
+
+	return buf
+}
+
+// SequenceId returns the sequence id attached to the InfoNotify event.
+// Events without a sequence number (KeymapNotify) return 0.
+// This is mostly used internally.
+func (v InfoNotifyEvent) SequenceId() uint16 {
+	return v.Sequence
+}
+
+// String is a rudimentary string representation of InfoNotifyEvent.
+func (v InfoNotifyEvent) String() string {
+	fieldVals := make([]string, 0, 5)
+	fieldVals = append(fieldVals, xgb.Sprintf("Sequence: %d", v.Sequence))
+	fieldVals = append(fieldVals, xgb.Sprintf("Timestamp: %d", v.Timestamp))
+	fieldVals = append(fieldVals, xgb.Sprintf("PowerLevel: %d", v.PowerLevel))
+	fieldVals = append(fieldVals, xgb.Sprintf("State: %t", v.State))
```

Comment:
We could just use a slice literal here?
```
fieldVals := []string{
	xgb.Sprintf("Sequence: %d", v.Sequence),
	…
}
```

Reply:
I tried to look to see if it was generated code. But I didn’t see it in this file.

Guess I should have looked at a different generated file. 🙈